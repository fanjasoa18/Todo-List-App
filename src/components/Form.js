import React , {useEffect} from "react";
import {v4 as uuisv4} from "uuid";

function Form({input, setInput, todos, setTodos, edit, setEdit}) {

    const updateTodo = (title, id, completed) => {
        const newTodo = todos.map((todo) => 
            todo.id === id ? {title, id, completed} : todo
        );
        setTodos(newTodo);
        setEdit("");
    };

    useEffect(() => {
        if(edit) {
            setInput(edit.title);
        } else {
            setInput("");
        }
    }, [setInput, edit]);

    const onInputChange = (event) => {
        setInput(event.target.value);
    };

    const onFormSubmit = (event) => {
        event.preventDefault();
        if(!edit) {
            setTodos([...todos, {id: uuisv4(), title: input, completed: false}]);
            setInput("");
        }else {
            updateTodo(input, edit.id,edit.completed)
        }

    };

    return(
       <form onSubmit={onFormSubmit} className="form-control">
            <input 
                type="text" 
                placeholder="Enter a Todo..." 
                className="task-input"
                value={input}
                required
                onChange={onInputChange}
            />
            <button type="submit" className="btn-add">
                {edit ? "OK" : "Add"}
            </button>
       </form> 
    )
}

export default Form;