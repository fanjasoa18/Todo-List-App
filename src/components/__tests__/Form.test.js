import React from 'react';
import { render, screen, cleanup, fireEvent } from '@testing-library/react';
import "@testing-library/jest-dom/extend-expect";
import {create} from "react-test-renderer";
import Form from '../Form';
afterEach(cleanup);

describe('Form', () => {

    it("renders correctly", () => {
        const onFormSubmit = jest.fn();
        render(<Form onSubmit={onFormSubmit}/>);

        const formElement = screen.getByTestId("form");
        expect(formElement).toBeInTheDocument();
        
    })

    test("snapshot renders correctly", () => {
        const onFormSubmit = jest.fn();
        const renderer = create(<Form onSubmit={onFormSubmit} />);
        const json = renderer.toJSON();

        expect(json).toMatchSnapshot();
    });


})