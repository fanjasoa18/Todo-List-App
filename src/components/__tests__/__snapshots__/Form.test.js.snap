// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Form snapshot renders correctly 1`] = `
<form
  className="form-control"
  data-testid="form"
  onSubmit={[Function]}
>
  <input
    className="task-input"
    data-testid="form-input"
    onChange={[Function]}
    placeholder="Enter a Todo..."
    required={true}
    type="text"
  />
</form>
`;